crates/proto/src/rr/rdata/caa.rs

```diff
@@ -890,6 +904,101 @@ impl fmt::Display for CAA {
     }
 }
 
+impl FromStr for CAA {
+    type Err = ProtoError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
This seems redundant to the existing parsing in the `txt` module of the `client` crate. I think we want to combine these and not duplicate the logic. Was there a reason you couldn't use that logic?

Reply:
I wasn't aware of that logic, I've never really used the client crate -- I guess I don't really understand what it's for? I've so far stuck with proto, resolver and server. Plus, it still seems weird that the server crate depends on the client crate, things like `LowerQuery` and `LowerName` feel like they belong in proto more than in client.

Reply:
Yes. I’m happy for us to shutdown the client crate at this point. I can review the history with you if you want, but at this point it does seem like the crate is a bit unnecessary. Moving it all into proto seems like a good idea. We could achieve the same goals of the crate by having a feature on proto.

Reply:
Yup, that sounds good to me.

Reply:
Yes. That’s fine. Move txt to proto.